### PR TITLE
(CDAP-1893) Fix a bug introduced when converting HBase queue state store...

### PR DESCRIPTION
**Pull request by chtyim**
... to dataset

### CODE REVIEW

**Comment by chtyim** [comment]
Working on adding a test case for that. But probably won't get in for 2.8.

File: cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/queue/AbstractQueueConsumer.java [hunk 1]
```diff
@@ -137,19 +137,11 @@ public ConsumerConfig getConfig() {
...
-          updateStartRow(floorKey);
-        }
-      } else {
-        // If the dequeue has empty result, startRow can advance to scanStartRow
-        updateStartRow(scanStartRow);
-      }
-    }
+    // Start row can be updated to the largest rowKey in the consumingEntries
+    // that is smaller than or equal to scanStartRow. If no such key exists, update start row to scanStartRow
+    byte[] floorKey = consumingEntries.floorKey(scanStartRow);
```
Comment (albertshau) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/queue/AbstractQueueConsumer.java]: old code checks that scanStartRow isn't null but this doesn't.  Has it changed to never be null?

**Comment by albertshau** [comment]
one question about possible NPE, otherwise lgtm

File: cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/queue/AbstractQueueConsumer.java (hunk 1 above)
Comment (chtyim) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/queue/AbstractQueueConsumer.java]: It never null now. It get initialized in constructor.

**Comment by chtyim** [comment]
Build passed: http://builds.cask.co/browse/CDAP-RBT201-1

**Pull request merged**